### PR TITLE
Add unit/integration tests to ci

### PR DESCRIPTION
**Pull request by ilackarms**

### CODE REVIEW

**Comment by mitchdraft** [comment]
changes to the nsutil look good to me

**Comment by ilackarms** [comment]
this PR is a summary of work it took to get the tests in `pkg/` passing:
- fix the NSUtil test by ensuring that namespace-resource maps are returned sorted (idempotent)
- fix dep / gopkg.toml
- add a small doc on enabling appmesh support (which was a necessary step for CI)
- switch from a custom type to the official prometheus config type for our prometheus syncer
- fix and improve appmesh syncer test

File: test/e2e/e2e_suite_test.go [hunk 1]
```diff
@@ -18,7 +18,7 @@ var PathToUds string
 
 var _ = BeforeSuite(func() {
 	var err error
-	PathToUds, err = gexec.Build("github.com/solo-io/solo-projects/projects/discovery/cmd")
+	PathToUds, err = gexec.Build("github.com/solo-io/gloo/projects/discovery/cmd")
```
Comment (mitchdraft) [inline test/e2e/e2e_suite_test.go]: there's one more of these solo-projects reference in `test/e2e/routing/routing_e2e_suite_test.go`

File: Gopkg.toml [hunk 2]
```diff
@@ -28,9 +28,9 @@
   go-tests = true
   unused-packages = true
 
-[[constraint]]
+[[override]]
   name = "github.com/solo-io/solo-kit"
-  version = "0.2.17"
+  revision = "554a82407c0b5d207722d3e0d5c668ac1a0622c8"
```
Comment (rickducott) [inline Gopkg.toml]: Switch to released version

**Comment by mitchdraft** [comment]
I'm getting some failures on `ginkgo -r` from the root directory @ilackarms 
```
Failed to compile e2e:

# github.com/solo-io/supergloo/test/e2e [github.com/solo-io/supergloo/test/e2e.test]
test/e2e/consul_e2e_test.go:131:34: too many arguments in call to "github.com/solo-io/supergloo/test/util".GetMeshClient
        have ("github.com/solo-io/supergloo/vendor/github.com/solo-io/solo-kit/pkg/api/v1/clients/kube".SharedCache)
        want ()
test/e2e/consul_e2e_test.go:132:42: too many arguments in call to "github.com/solo-io/supergloo/test/util".GetUpstreamClient
        have ("github.com/solo-io/supergloo/vendor/github.com/solo-io/solo-kit/pkg/api/v1/clients/kube".SharedCache)
        want ()
test/e2e/istio_e2e_test.go:134:34: too many arguments in call to "github.com/solo-io/supergloo/test/util".GetMeshClient
        have ("github.com/solo-io/supergloo/vendor/github.com/solo-io/solo-kit/pkg/api/v1/clients/kube".SharedCache)
        want ()
test/e2e/istio_e2e_test.go:135:42: too many arguments in call to "github.com/solo-io/supergloo/test/util".GetUpstreamClient
        have ("github.com/solo-io/supergloo/vendor/github.com/solo-io/solo-kit/pkg/api/v1/clients/kube".SharedCache)
        want ()
```

File: test/util/test_helpers.go [hunk 3]
```diff
@@ -291,32 +294,28 @@ func WaitForDeletedPods(namespace string) {
 	WaitForDeletedPodsWithTimeout(namespace, "120s")
 }
 
-func GetMeshClient(kubeCache kube.SharedCache) v1.MeshClient {
+func GetMeshClient() v1.MeshClient {
```
Comment (mitchdraft) [inline test/util/test_helpers.go]: I believe this is breaking some tests

Comment (ilackarms) [reply test/util/test_helpers.go]: the e2e tests are failing for other reasons and will be fixed in a followup PR
all the tests in `pkg` should be passing

**Pull request merged**